hack/generate.sh

```diff
@@ -71,9 +72,32 @@ function generate_configuration() {
     sed -i "s|OPENSHIFT_VERSIONS=.*|OPENSHIFT_VERSIONS=${OPENSHIFT_VERSIONS}|" ${__root}/config/onprem-iso-fcc.yaml
     sed -i "s|PUBLIC_CONTAINER_REGISTRIES=.*|PUBLIC_CONTAINER_REGISTRIES=${PUBLIC_CONTAINER_REGISTRIES}|" ${__root}/config/onprem-iso-fcc.yaml
 
-    sed -i "s|value: '.*' # openshift version|value: '${OPENSHIFT_VERSIONS}' # openshift version|" ${__root}/config/manager/manager.yaml
     sed -i "s|HW_VALIDATOR_REQUIREMENTS=.*|HW_VALIDATOR_REQUIREMENTS=${HW_VALIDATOR_REQUIREMENTS}|" ${__root}/config/onprem-iso-fcc.yaml
     docker run --rm -v ${__root}/config/onprem-iso-fcc.yaml:/config.fcc:z quay.io/coreos/fcct:release --pretty --strict /config.fcc > ${__root}/config/onprem-iso-config.ign
+
+    # Updated operator manifests with openshift versions
+    sed -i "s|value: '.*' # openshift version|value: '${OPERATOR_OPENSHIFT_VERSIONS}' # openshift version|" ${__root}/config/manager/manager.yaml
+    # This python is responsible for updating the sample AgentServiceConfig to include the latest + correct osImages
+    # When the CSV is built, this is included in the `almExamples` so that when a user goes through the OpenShift console
+    # to create the `agent` this will give them the correct defaults.
+    python3 -c '
```

Comment:
Would using `jq` to generate the desired json and then `sed` work? 

This is a snippet with a `jq` filter that should the same thing this python script does:

```
echo '{"4.8":{"display_name":"4.8.0-fc.3","release_version":"4.8.0-fc.3","release_image":"quay.io/openshift-release-dev/ocp-release:4.8.0-fc.3-x86_64","rhcos_image":"https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/pre-release/4.8.0-fc.3/rhcos-4.8.0-fc.3-x86_64-live.x86_64.iso","rhcos_rootfs":"https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/pre-release/4.8.0-fc.3/rhcos-live-rootfs.x86_64.img","rhcos_version":"48.84.202105062123-0","support_level":"beta"}}' | jq '[. | to_entries | .[] | {"openshiftVersion": .key, "version": .value["rhcos_version"], "url": .value["rhcos_image"], "rootFSUrl": .value["rhcos_rootfs"] }]'
[
  {
    "openshiftVersion": "4.8",
    "version": "48.84.202105062123-0",
    "url": "https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/pre-release/4.8.0-fc.3/rhcos-4.8.0-fc.3-x86_64-live.x86_64.iso",
    "rootFSUrl": "https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/pre-release/4.8.0-fc.3/rhcos-live-rootfs.x86_64.img"
  }
]
```

You can play with it here: https://jqplay.org/s/ABfiwfFacJ

Reply:
is there a way to transform the json to yaml ?

Reply:
Technically JSON is valid YAML

Reply:
> is there a way to transform the json to yaml ?

Plenty, but let's not hold this PR on that. 

Personally, I think this snippet is not very maintainable, or readable. I would like us to use the tools available in the dev environments/systems we have.

> Technically JSON is valid YAML

@omertuc I am sure @djzager knows this :smile: the question was related to having the example be fully `YAML` as his script is currently generating it. In your defense, @djzager's comment was very succint. 

Let's try to provide more context in these comments to help conversations move forward